saleor/graphql/discount/tests/test_voucher_channel_listing.py

```diff
@@ -0,0 +1,577 @@
+import graphene
+
+from ....discount.error_codes import DiscountErrorCode
+from ...tests.utils import assert_no_permission, get_graphql_content
+
+VOUCHER_CHANNEL_LISTING_UPDATE_MUTATION = """
+mutation UpdateVoucherChannelListing($id: ID!, $input: VoucherChannelListingInput!) {
+    voucherChannelListingUpdate(id: $id, input: $input) {
+        discountErrors {
+            field
+            message
+            code
+            channels
+        }
+        voucher {
+            code
+            channelListing {
+                channel{
+                    slug
+                }
+                discountValue
+                currency
+                minSpent {
+                    currency
+                    amount
+                }
+            }
+        }
+    }
+}
+"""
+
+
+def test_voucher_channel_listing_create_as_staff(
+    staff_api_client, voucher_without_channel, permission_manage_discounts, channel_USD
+):
+    # given
+    voucher = voucher_without_channel
+    voucher_id = graphene.Node.to_global_id("Voucher", voucher.pk)
+    channel_id = graphene.Node.to_global_id("Channel", channel_USD.id)
+    variables = {
+        "id": voucher_id,
+        "input": {"addChannels": [{"channelId": channel_id, "discountValue": 5.5}]},
+    }
+
+    # when
+    response = staff_api_client.post_graphql(
+        VOUCHER_CHANNEL_LISTING_UPDATE_MUTATION,
+        variables=variables,
+        permissions=(permission_manage_discounts,),
+    )
+    content = get_graphql_content(response)
+
+    # then
+    data = content["data"]["voucherChannelListingUpdate"]["voucher"]
+    assert not content["data"]["voucherChannelListingUpdate"]["discountErrors"]
+    channel_listing = data["channelListing"]
+    assert channel_listing[0]["channel"]["slug"] == channel_USD.slug
+    assert channel_listing[0]["discountValue"] == 5.5
+    assert channel_listing[0]["currency"] == channel_USD.currency_code
+
+
+def test_product_channel_listing_update_as_app(
+    app_api_client, voucher_without_channel, permission_manage_discounts, channel_USD
+):
+    # given
+    voucher = voucher_without_channel
+    voucher_id = graphene.Node.to_global_id("Voucher", voucher.pk)
+    channel_id = graphene.Node.to_global_id("Channel", channel_USD.id)
+    variables = {
+        "id": voucher_id,
+        "input": {"addChannels": [{"channelId": channel_id, "discountValue": 5.5}]},
```

Comment:
Small improvement, if you are reusing this value, it will be better to store this value in variable.